.github/workflows/release.yml

```diff
@@ -1,16 +1,20 @@
 name: Release
 
-permissions:
-  contents: write
-
 on:
   push:
     tags:
       - "v*.*.*"
+    branches:
+      # NOTE: make sure any branches here are also valid directory names,
+      # otherwise creating the directory and uploading to s3 will fail
+      - 'main'
 
 jobs:
   release:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write # In order to upload artifacts to GitHub releases
```

Comment:
So I'm not sure how safe it is to give these permissions in the normal CI runs, could we make a new job that just runs on the main tags and has this permission that does the publish only?

Reply:
How's this? [`105798c` (#152)](https://github.com/DeterminateSystems/nix-installer/pull/152/commits/105798cea83984117d3dc67f47cdc13f3f877b09)